packages/block-library/src/query/edit/inspector-controls/order-control.js

```diff
@@ -4,34 +4,14 @@
 import { SelectControl } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 
-const orderOptions = [
-	{
-		label: __( 'Newest to oldest' ),
-		value: 'date/desc',
-	},
-	{
-		label: __( 'Oldest to newest' ),
-		value: 'date/asc',
-	},
-	{
-		/* translators: Label for ordering posts by title in ascending order. */
-		label: __( 'A → Z' ),
-		value: 'title/asc',
-	},
-	{
-		/* translators: Label for ordering posts by title in descending order. */
-		label: __( 'Z → A' ),
-		value: 'title/desc',
-	},
-];
-function OrderControl( { order, orderBy, onChange } ) {
+function OrderControl( { order, orderBy, orderByOptions, onChange } ) {
```

Comment:
We keep this component decoupled from any editor state and expect to receive the `orderByOptions` from the user similar to the `order` and `orderBy`.

**Question**: Is this a breaking change and should we consider an upgrade path for anyone using this component? Now, without the `orderByOptions` value it will render an empty selector.

Reply:
@kasparsd, we could leave the `orderOptions` definition and use it as the default when no `orderByOptions` is provided.

Reply:
That's a great suggestion -- I've added this back as the default value for the argument.